hammer/config/defaults.yml

```diff
@@ -613,6 +613,9 @@ par:
       strap_layers: [] # List of layers on which to place power straps (std cell rail layer is implied - do not include it)
       # type: List[str]
 
+      generate_rail_layer: true # If true, generates straps on the stdcell rail layer - technology.core.std_cell_rail_layer
```

Comment:
I'd prefer this to be one level up in the key hierarchy (i.e. `par.generate_power_straps_options.generate_rail_layer`). This is because the rail layer generation doesn't use any of the `by_tracks` calculations.

Reply:
Makes sense. Following that reasoning,  should we also pull `self.specify_std_cell_power_straps` out of `specify_all_power_straps_by_tracks` into `generate_power_straps_tcl`?

Reply:
Yes, let's do that.

Reply:
I see that the  fields of `generate_power_straps_options` are intended to be the valid options for `generate_power_straps_method`. Do you think its a good idea to add another field to `generate_power_Straps_options` that breaks that pattern? I don't think it would cause anything to break, but it does invalidate that comment.

```
  generate_power_straps_method: "by_tracks" # If power_straps_mode is 'generate', which method to use.
  # Currently, the valid options are:
  # - by_tracks - Specify the power strap plan per layer in terms of tracks

  generate_power_straps_options:
  # Default settings for power strap generation modes
  # Keys are the valid values of the 'generate_power_straps_method' key above
    by_tracks:
    ```

Reply:
Ah, I see. Yeah, let's not break that pattern then. This PR is fine as-is.